kernel/Kconfig.power_mgmt

```diff
@@ -59,6 +60,8 @@ config SYS_POWER_DEEP_SLEEP
 	  from the reset vector same as cold boot. The interface allows
 	  restoration of states that were saved at the time of suspend.
 
+endif # SYS_POWER_MANAGEMENT
```

Comment:
Moving this up here is unrelated to the commit, what is the reasoning? device pm can now work on its own without system power management?

Reply:
> Moving this up here is unrelated to the commit, what is the reasoning?

Without the change any project using device power management would automatically be configured in tickles idle mode. That's because enabling device power management would only be possible if system power management is enabled. Since these two features are independent we allow projects to use device power management without tickles idle as it was the case before.

> device pm can now work on its own without system power management?

As far as I can tell looking at the source code device power management is independent from system power management. If it is not the case then this change should not be performed.